stdlib/public/runtime/Metadata.cpp

```diff
@@ -673,6 +700,42 @@ swift::swift_allocateGenericValueMetadata(const ValueTypeDescriptor *description
   return metadata;
 }
 
+MetadataResponse swift::swift_getCanonicalSpecializedMetadata(
+    MetadataRequest request, const Metadata *candidate,
+    const Metadata **cacheMetadataPtr) {
+  assert(candidate->isStaticallySpecializedGenericMetadata() &&
+         !candidate->isCanonicalStaticallySpecializedGenericMetadata());
+  auto *description = candidate->getDescription();
+  assert(description);
+
+  using CachedMetadata = std::atomic<const Metadata *>;
+  auto cachedMetadataAddr = ((CachedMetadata *)cacheMetadataPtr);
+  auto *cachedMetadata = cachedMetadataAddr->load(std::memory_order_acquire);
```

Comment:
We should be able to use SWIFT_MEMORY_ORDER_CONSUME here instead of acquire, since any writes to the metadata from other threads should be accessed by this thread only through the returned pointer.

Reply:
Thank you for the explanation!  Fixed.